plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -171,33 +174,47 @@ func createClient(s *options.SchedulerServer) (*clientset.Clientset, error) {
 	return cli, nil
 }
 
-func createConfig(s *options.SchedulerServer, kubecli *clientset.Clientset) (*scheduler.Config, error) {
-	configFactory := factory.NewConfigFactory(kubecli, s.SchedulerName, s.HardPodAffinitySymmetricWeight, s.FailureDomains)
-	if _, err := os.Stat(s.PolicyConfigFile); err == nil {
+// schedulerConfigurator is an interface wrapper that provides default Configuration creation based on user
+// provided config file.
+type schedulerConfigurator struct {
+	scheduler.Configurator
+	policyFile        string
+	algorithmProvider string
+}
+
+func (sc schedulerConfigurator) Create() (*scheduler.Config, error) {
+	// Try to create a valid configuration.
+	if _, err := os.Stat(sc.policyFile); err == nil {
 		var (
 			policy     schedulerapi.Policy
 			configData []byte
 		)
-		configData, err := ioutil.ReadFile(s.PolicyConfigFile)
+		configData, err := ioutil.ReadFile(sc.policyFile)
 		if err != nil {
 			return nil, fmt.Errorf("unable to read policy config: %v", err)
 		}
 		if err := runtime.DecodeInto(latestschedulerapi.Codec, configData, &policy); err != nil {
 			return nil, fmt.Errorf("invalid configuration: %v", err)
 		}
-		return configFactory.CreateFromConfig(policy)
-	}
+		return sc.Configurator.CreateFromConfig(policy)
 
-	// if the config file isn't provided, use the specified (or default) provider
-	config, err := configFactory.CreateFromProvider(s.AlgorithmProvider)
-	if err != nil {
-		return nil, err
+	} else {
```

Comment:
I would change the order of this if to avoid else. How about:
```
if _, err := os.Stat(sc.policyFile); err != nil {
  return sc.Configurator.CreateFromProvider(sc.algorithmProvider
}
configData, err := ...
...
```
?